flink-runtime/src/main/java/org/apache/flink/runtime/registration/RegisteredRpcConnection.java

```diff
@@ -91,9 +92,15 @@ public void start() {
 
 		future.whenCompleteAsync(
 			(Tuple2<G, S> result, Throwable failure) -> {
-				// this future should only ever fail if there is a bug, not if the registration is declined
 				if (failure != null) {
-					onRegistrationFailure(failure);
+					if (failure instanceof CancellationException) {
+						// we ignore cancellation exceptions because the originate from cancelling
```

Comment:
the -> they

Reply:
Thanks for spotting the error. Will fix it.